### PR TITLE
add Approximating non function mapping example

### PR DESCRIPTION
**Pull request by LukeWood**
This guide covers how to approximate non-function one to many mappings using Keras Core: https://colab.research.google.com/drive/1kRYIn8lMuvK8wc9uloEwJvZ7foTUNPnV?usp=sharing

PTAL!

### CODE REVIEW

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
I put a bunch of fixes in this PR:

https://github.com/LukeWood/keras-io/pull/2, Please merge, lgtm.

I wasn't able to push it directly to your branch. 

Do new files need to be added to index pages for keras.io?

File: examples/keras_recipes/approximating_non_function_mappings.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+Accelerator: CPU
+"""
+
+"""
+## Approximating NonFunctions
+
+Neural networks are universal function approximators. Key word: function!
+While powerful function approximators, neural networks are not able to
+approximate non-functions.
+One important restriction to remember about functions - they have one input, one
```
Comment (LukeWood) [inline examples/keras_recipes/approximating_non_function_mappings.py:17]: Perhaps...

One important characteristic of functions is that they map one input to a one unique output.
Neural networks suffer greatly when the training set has multiple values of Y for a single X.

File: examples/keras_recipes/approximating_non_function_mappings.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+y = np.concatenate(ys, axis=0)
+x = np.expand_dims(x, axis=1)
+
+plt.scatter(x, y)
+plt.show()
+
+"""
+As you can see, there's multiple possible values for Y with respect to a given
+X.  Normal neural networks will simply learn the mean of these points with
+respect to geometric space.
```
Comment (LukeWood) [inline examples/keras_recipes/approximating_non_function_mappings.py:105]: However, in the context of our spiral the geometric mean of the each Y occurs with a probability of zero...

File: examples/keras_recipes/approximating_non_function_mappings.py [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+plt.scatter(x, y)
+plt.scatter(x, y_pred)
+plt.show()
+
+"""
+
+## Mixture Density Networks
+
+Mixture Density networks can alleviate this problem.
+A Mixture density is a class of complicated densities expressible in terms of simpler densities.
```
Comment (LukeWood) [inline examples/keras_recipes/approximating_non_function_mappings.py:158]: Mixture density -> mixture density

File: examples/keras_recipes/approximating_non_function_mappings.py [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+They are effectively the sum of a ton of probability distributions.
```
Comment (LukeWood) [inline examples/keras_recipes/approximating_non_function_mappings.py:159]: effectively, a mixture density is the sum of various probability distributions.  By summing various distributions, mixture densitry distributions can effectively model arbitrarily complex distributions.

**Review by LukeWood — COMMENTED** [review]
@fchollet I think this looks pretty good - @MarkDaoust added some edits.

I left a few final copy-edits on my own wording which I can apply later today - but let me know if this is a good addition.

Edit: sorry, misunderstood Mark's comment.  Merged his PR on my repo.

**Comment by LukeWood** [comment]
@MarkDaoust I merged them - let me know if anything else is needed!

**Comment by LukeWood** [comment]
FYI @MarkDaoust you may want to re-generate the markdown, the learned MDN is pretty poor here - and I've been able to get much stronger convergences.  Maybe we can just run it 1-2 more times to get good figures

**Review by MarkDaoust — APPROVED** [review]
LGTM.

**Comment by MarkDaoust** [comment]
@LukeWood did you want to add those copy edits and regenerate, or should I just merge this?

**Comment by LukeWood** [comment]
Go ahead @MarkDaoust - I think the point is still clear and I'm happy with it.  Cheers!

**Pull request merged**